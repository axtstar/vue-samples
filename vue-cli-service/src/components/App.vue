<template>
  <div id="app">
    <h1>Shopping Cart Example</h1>
    <hr>
    <h2>Products</h2>
    <ProductList/>
    <hr>
    <ShoppingCart/>
    <hr>
    <Counter/>
  </div>
</template>

<script>
import ProductList from './ProductList.vue'
import ShoppingCart from './ShoppingCart.vue'
import Counter from './Counter.vue'
export default {
  components: { ProductList, ShoppingCart, Counter }
}
</script>