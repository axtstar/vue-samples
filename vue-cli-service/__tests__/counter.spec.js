import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import Vuex from 'vuex'
import App from '../src/components/Counter.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Counter.vue', () => {
  let actions
  let store

  // コンポーネントがマウントされ、ラッパが作成されます。
  const wrapper = mount(App)

  // wrapper.vmを 介して実際の Vue インスタンスにアクセスできます
  const vm = wrapper.vm

  console.log(wrapper.html())

  it('vue contains div', ()=>{
    // div
    expect(wrapper.find('div').exists()).toBeTruthy()
  })

  it('add 1 after button click', ()=>{
    expect(wrapper.vm.count).toBe(0)
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.count).toBe(1)
    expect(wrapper.element).toMatchSnapshot()
  })

})
