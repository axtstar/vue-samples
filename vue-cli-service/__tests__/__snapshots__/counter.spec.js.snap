// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue add 1 after button click 1`] = `
<div
  id="counter"
>
  <div>
    カウンタ
  </div>
   
  <span
    class="count"
  >
    0
  </span>
   
  <button>
    Increment
  </button>
</div>
`;
